semgrep/semgrep/config_resolver.py

```diff
@@ -119,7 +119,7 @@ def __init__(self, config_str: str, project_url: Optional[str]) -> None:
             self._config_path = f"{SEMGREP_URL}{AUTO_CONFIG_LOCATION}"
         else:
             self._origin = ConfigType.LOCAL
-            self._config_path = config_str
+            self._config_path = str(Path(config_str).expanduser())
```

Comment:
Should we add a test?

Reply:
Yes but a little difficult with current test framework. Probably fine for now :D